controllers/confidential_handler.go

```diff
@@ -57,19 +71,49 @@ func (r *KataConfigOpenShiftReconciler) handleFeatureConfidential(state FeatureG
 	}
 
 	// peer pods config
+	if r.kataConfig.Spec.EnablePeerPods {
+		// Patch peer pods configMap, if it exists.
+		var peerpodsCMData map[string]string
+		if state == Enabled {
+			peerpodsCMData = map[string]string{"DISABLECVM": "false"}
+		} else {
+			peerpodsCMData = map[string]string{"DISABLECVM": "true"}
+		}
+		if err := updateConfigMap(r.Client, r.Log, peerpodsCMName, OperatorNamespace, peerpodsCMData); err != nil {
+			if k8serrors.IsNotFound(err) {
+				// When feature is Enabled: ConfigMap doesn't exist yet, will try again at the next reconcile run.
+				// Else: Nothing to do, feature is disabled and configMap doesn't exist.
+			} else {
+				return err
+			}
+		}
+	}
 
-	// Patch peer pods configMap, if it exists.
-	var peerpodsCMData map[string]string
+	// Handle kata-cc runtime classes for baremetal environments
 	if state == Enabled {
-		peerpodsCMData = map[string]string{"DISABLECVM": "false"}
+		r.Log.Info("Creating kata-cc runtime classes for confidential containers")
+
+		// Create kata-cc-intel runtime class for Intel TDX
+		err := r.createRuntimeClass(kataCCIntelRuntimeClassName, kataCCIntelRuntimeClassCpuOverhead, kataCCIntelRuntimeClassMemOverhead, intelTDXNodeLabel)
+		if err != nil {
+			r.Log.Info("Error creating kata-cc-intel runtime class", "err", err)
+			return err
+		}
+
+		// Create kata-cc-amd runtime class for AMD SNP
+		err = r.createRuntimeClass(kataCCAMDRuntimeClassName, kataCCAMDRuntimeClassCpuOverhead, kataCCAMDRuntimeClassMemOverhead, amdSNPNodeLabel)
```

Comment:
Do we really want to create arch specific runtimeClass or just use `kata-cc` as the runtimeClass with handlers as `kata-cc-{arch}` ?

Reply:
I'm ok with that approach. 

However, @gkurz mentioned something about having multiple runtimeclasses but also having a single one to be the "smart" one. Its still not clear to me, if we really need those, if the users can choose the nodes by using the node affinity labels.

So I will just wait a bit for his comments, to see if he has something against this.